javascript/ql/lib/semmle/javascript/heuristics/AdditionalSources.qll

```diff
@@ -52,3 +52,30 @@ class RemoteServerResponse extends HeuristicSource, RemoteFlowSource {
 
   override string getSourceType() { result = "a response from a remote server" }
 }
+
+/**
+ * The data read from a database.
+ */
+class DatabaseAccessResultRemoteFlowSource extends HeuristicSource, RemoteFlowSource {
+  DatabaseAccessResultRemoteFlowSource() { exists(DatabaseAccess dba | this = dba.getAResult()) }
+
+  override string getSourceType() { result = "Database query result" }
+
+  override predicate isUserControlledObject() { any() }
+}
+
+/**
+ * A remote flow source originating from a database access.
+ */
+private class RemoteFlowSourceFromDBAccess extends RemoteFlowSource, HeuristicSource {
+  RemoteFlowSourceFromDBAccess() {
+    this = ModelOutput::getASourceNode("database-access-result").getAUse()
```

Comment:
```suggestion
    this = ModelOutput::getASourceNode("database-access-result").getAUse()
    or
    exists(DatabaseAccess dba | this = dba.getAResult())
```

One last comment, couldn't you combine the two new classes in `AdditionalSources.qll`?

Reply:
sure, done